### PR TITLE
swagger: update api docs to include vehicle location api endpoints

### PR DESCRIPTION
**Pull request by ehzhang**
image
image

### CODE REVIEW

File: swagger.json [hunk 1]
```diff
@@ -1691,6 +1741,64 @@
...
+              "name": "vehicle_id",
+              "in": "path",
+              "required": true,
+              "description": "ID of the vehicle with the associated routes.",
+              "type": "integer",
+              "format": "int64"
+            },
+            {
+              "name": "startMs",
+              "description": "Time in millis for the start of the query (cannot exceed 1 hour)",
```
Comment (silverspace) [inline swagger.json]: The limit is 30 minutes, not 1 hour, correct?

File: swagger.json [hunk 2]
```diff
@@ -2294,6 +2402,65 @@
...
+            },
+            "speedMilesPerHour": {
+              "type": "number",
+              "description": "The speed calculated from GPS that the asset was traveling at in miles per hour.",
+              "example": "35"
+            },
+            "timeMs": {
+              "type": "number",
+              "description": "Time in Unix milliseconds since epoch when the asset was at the location.",
+              "example": "12314151"
```
Comment (silverspace) [inline swagger.json]: Could we use an actual unixMs example?

File: swagger.json [hunk 3]
```diff
@@ -2294,6 +2402,65 @@
...
+            },
+            "longitude": {
+              "type": "number",
+              "description": "The longitude of the location in degrees.",
+              "example": "-122"
+            },
+            "speedMilesPerHour": {
+              "type": "number",
+              "description": "The speed calculated from GPS that the asset was traveling at in miles per hour.",
+              "example": "35"
```
Comment (silverspace) [inline swagger.json]: We should update this to be 35.3 to illustrate that it's floating point and not integer.

File: swagger.json [hunk 4]
```diff
@@ -2294,6 +2402,65 @@
...
+              "description": "The latitude of the location in degrees.",
+              "example": "37"
+            },
+            "longitude": {
+              "type": "number",
+              "description": "The longitude of the location in degrees.",
+              "example": "-122"
+            },
+            "speedMilesPerHour": {
+              "type": "number",
```
Comment (silverspace) [inline swagger.json]: We should also add `"format": "double"` to all of our lat/lng/speed parameters.

File: swagger.json [hunk 5]
```diff
@@ -2294,6 +2402,65 @@
...
+              "description": "The speed calculated from GPS that the asset was traveling at in miles per hour.",
+              "example": "35"
+            },
+            "timeMs": {
+              "type": "number",
```
Comment (silverspace) [inline swagger.json]: Could we use `"format": "int64"` here? 
https://swagger.io/docs/specification/data-models/data-types/#numbers

File: swagger.json [hunk 6]
```diff
@@ -1691,6 +1741,64 @@
...
+            "operationId": "getVehicleLocations",
+            "consumes": [
+              "application/json"
+            ],
+            "produces": [
+              "application/json"
+            ],
+            "responses": {
+              "200": {
+                "description": "Returns locations for a given vehicle between a start/end time, with a maximum query duration of one hour.",
```
Comment (silverspace) [inline swagger.json]: Is the max query duration 30 minutes?

File: swagger.json [hunk 7]
```diff
@@ -1691,6 +1741,64 @@
...
+            {
+              "name": "startMs",
+              "description": "Time in millis for the start of the query (cannot exceed 1 hour)",
+              "required": true,
+              "in": "query",
+              "type": "integer"
+            },
+            {
+              "name": "endMs",
+              "description": "Time in millis for the end of the query (cannot exceed 1 hour)",
```
Comment (silverspace) [inline swagger.json]: `Time in Unix millis`.

File: swagger.json (hunk 6 above)
Comment (ehzhang) [reply swagger.json]: It is 1 hour for the single vehicle case, but 30 minutes for all vehicles

File: swagger.json (hunk 4 above)
Comment (ehzhang) [reply swagger.json]: do you mean float?

File: swagger.json (hunk 3 above)
Comment (ehzhang) [reply swagger.json]: sg

File: swagger.json (hunk 5 above)
Comment (ehzhang) [reply swagger.json]: sure - i copy pasted this from another set of args 😬

File: swagger.json [hunk 8]
```diff
@@ -1691,6 +1741,64 @@
...
+              "in": "path",
+              "required": true,
+              "description": "ID of the vehicle with the associated routes.",
+              "type": "integer",
+              "format": "int64"
+            },
+            {
+              "name": "startMs",
+              "description": "Time in millis for the start of the query (cannot exceed 1 hour)",
+              "required": true,
```
[the remark this answers is not in the document]
Comment (ehzhang) [reply swagger.json]: The backend does not have defaults - I think that might be nice, but I'm also pretty okay with having them required every time.

File: swagger.json (hunk 2 above)
Comment (ehzhang) [reply swagger.json]: yes! bad copy paste

Comment (ehzhang) [reply swagger.json]: yes

File: swagger.json (hunk 3 above)
Comment (ehzhang) [reply swagger.json]: done

File: swagger.json (hunk 4 above)
Comment (ehzhang) [reply swagger.json]: changed to float

File: swagger.json (hunk 5 above)
Comment (ehzhang) [reply swagger.json]: changed to include format

File: swagger.json (hunk 7 above)
Comment (ehzhang) [reply swagger.json]: done

**Pull request merged**